File: 3-leetcode/01-easy/js/2022-convert-1d-array-into-2d-array.ts
const construct2DArray = (
  original: number[],
  m: number,
  n: number,
): number[][] => {
  if (original.length != m * n) {
    return [];
  }

  const result = Array.from({ length: m }, () =>
    Array.from({ length: n }, () => 0),
  );

  for (let i = 0; i < original.length; i++) {
    result[Math.floor(i / n)][i % n] = original[i];
  }

  return result;
};

File: 3-leetcode/01-easy/dart/2022-convert-1d-array-into-2d-array.dart
class Solution {
  List<List<int>> construct2DArray(List<int> original, int m, int n) {
    if (original.length != m * n) {
      return [];
    }

    List<List<int>> result =
        List.generate(m, (i) => List.generate(n, (i) => 0));

    for (int i = 0; i < original.length; i++) {
      result[(i ~/ n)][i % n] = original[i];
    }

    return result;
  }
}
